packages/contracts/src.ts/index.ts

```diff
@@ -224,22 +219,19 @@ async function populateTransaction(contract: Contract, fragment: FunctionFragmen
 
     // The ABI coded transaction
     const data = contract.interface.encodeFunctionData(fragment, resolved.args);
-    const tx: PopulatedTransaction = {
-      data: data,
-      to: resolved.address
+    const tx: TransactionRequest = {
+        data: data,
+        to: resolved.address
     };
 
     // Resolved Overrides
     const ro = resolved.overrides;
 
     // Populate simple overrides
-    if (ro.nonce != null) { tx.nonce = BigNumber.from(ro.nonce).toNumber(); }
     if (ro.gasLimit != null) { tx.gasLimit = BigNumber.from(ro.gasLimit); }
-    if (ro.gasPrice != null) { tx.gasPrice = BigNumber.from(ro.gasPrice); }
     if (ro.maxFeePerGas != null) { tx.maxFeePerGas = BigNumber.from(ro.maxFeePerGas); }
     if (ro.maxPriorityFeePerGas != null) { tx.maxPriorityFeePerGas = BigNumber.from(ro.maxPriorityFeePerGas); }
     if (ro.from != null) { tx.from = ro.from; }
-
     if (ro.type != null) { tx.type = ro.type; }
     if (ro.accessList != null) { tx.accessList = accessListify(ro.accessList); }
 
```

Comment:
The calculations for the intrinsic gas costs are slightly different in Hedera:
https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/contracts/gascalculator/GasCalculatorHederaV22.java

In ethereum, the whitepaper states that:
there is 21k gas at minimum charged + for every `0` byte we charge `4` gas + for every non-zero byte we charge `64` gas 

however, in Hedera the formula is the following:
- 21k is charged + for every `0` byte we charge `4` gas + for every non-zero byte we charge `16` bytes + if it is contract create, we charge 32k gas